packages/babel-plugin-proposal-object-rest-spread/test/fixtures/object-rest/nested-computed-key/output.js

```diff
@@ -1,7 +1,7 @@
 var _ref2;
 
 const {
-  [(_ref) => {
+  [_ref => {
```

Comment:
I looked into why some of these arrow functions are *losing* their parens.

It turns out that the previous implementation of `hasTypes` was checking `param.trailingComments`, which is truthy when `trailingComments` is an empty array.

Switching to `param.trailingComments?.length` makes the generator conclude that this arrow function doesn't need parentheses after all.